src/Underscores.jl

```diff
@@ -109,6 +118,8 @@ The detailed rules are:
    expands the closure scope to the whole expression.
 4. Piping and composition chains with `|>,<|,∘` are treated as a special case
    where the replacement recurses into sub-expressions.
+5. Indexing `[...]` is a similar special case as regards `_` (and `_1,_2,...`),
+   but does not affect `__`.
```

Comment:
I think we'd best fold this into rule 1 and just say that `_` is passed to "the outermost parenthesized expression [*]".  Though that implies a larger rewrite which affects things other than `Expr(:ref)`.

Of course, the difficulty is that we can't guarantee this behavior because the parser doesn't tell us about it. But I think we should strive to come as close as we can, and document the caveats in a separate block (the `[*]`)?

Reply:
OK, this indeed sounds better. I've moved this down to a paragraph in Extended Help. See what you think. 

It has a one-item table, but with the idea that other things may arrive. Such as a rule excluding infix operators, possibly just making them errors?